services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crossregion/utils/CrossRegionUtils.java

```diff
@@ -59,7 +61,16 @@ public static boolean isS3RedirectException(Throwable exception) {
     }
 
     private static boolean isRedirectError(S3Exception exceptionToBeChecked) {
-        if (REDIRECT_STATUS_CODES.stream().anyMatch(status -> status.equals(exceptionToBeChecked.statusCode()))) {
+        boolean matchRedirectStatusCode =
+            REDIRECT_STATUS_CODES.stream()
+                                 .anyMatch(status -> status.equals(exceptionToBeChecked.statusCode()));
+        if (matchRedirectStatusCode) {
+            return true;
+        }
+        boolean is400IllegalLocationConstraintException =
+            exceptionToBeChecked.statusCode() == ILLEGAL_LOCATION_CONSTRAINT_EXCEPTION_STATUS_CODE
```

Comment:
Can we just use 400 here? 400 is easier to read than ILLEGAL_LOCATION_CONSTRAINT_EXCEPTION_STATUS_CODE IMO...